routes/pages.js

```diff
@@ -18,10 +18,9 @@ router.get('/about', (req, res) => {
 
 router.get('/movies/:id', async (req, res) => {
   const movieRender = await specMovie(req.params.id);
-  if (movieRender === 'Page not found') {
-    const errorText = 'Filmen kunde inte hittas!'
+  if (movieRender === '404 (Not found)') {
     res.status(404)
-    res.render('movies', { movieRender: errorText, title: 'Page not found', sectionTitle: '' })
+    res.render('movies', { movieRender: movieRender, title: movieRender, sectionTitle: '' })
```

Comment:
Here you're using the same template (`'movies'`) for all pages, regardless of whether it's the single-movie page, the movies list, or the 404 message. That requires a lot of logic in the template.

Instead, I would use three different templates for this, one called `movies`, one called `movie`, and one called `404`.